src/auxiliary/auxiliary.jl

```diff
@@ -38,18 +38,20 @@ end
 
 PerformanceCounter() = PerformanceCounter(0, 0.0)
 
-function Base.take!(counter::PerformanceCounter)
+@inline function Base.take!(counter::PerformanceCounter)
   time_per_call = counter.runtime / counter.ncalls_since_readout
   counter.ncalls_since_readout = 0
   counter.runtime = 0.0
   return time_per_call
 end
 
-function Base.put!(counter::PerformanceCounter, runtime::Real)
+@inline function Base.put!(counter::PerformanceCounter, runtime::Real)
```

Comment:
I'm fine with these changes. Just out of curiosity - were they necessary?

Reply:
Not sure. I just thought that since these functions need to be as fast as possible, it makes sense to encourage the compiler to inline them. I didn't check it though, and can revert it if you'd like.

Reply:
No, it's fine.